compute/metadata/metadata.go

```diff
@@ -304,8 +306,20 @@ func (c *Client) getETag(suffix string) (value, etag string, err error) {
 	}
 	req.Header.Set("Metadata-Flavor", "Google")
 	req.Header.Set("User-Agent", userAgent)
-	res, err := c.hc.Do(req)
-	if err != nil {
+	var res *http.Response
+	retryer := newRetryer()
+	for {
+		var err error
+		res, err = c.hc.Do(req)
+		if err == nil {
+			break
+		}
+		if delay, shouldRetry := retryer.Retry(res.StatusCode, err); shouldRetry {
+			if err := gax.Sleep(context.Background(), delay); err != nil {
```

Comment:
This seems like we should declare a `ctx := context.TODO()` or something at the root of the function.

Reply:
An alternative would be putting the timeout in here and removing the retry counting from the retryer. Not sure what is better.

Reply:
TODO it is!